readme.md

```diff
@@ -420,26 +420,26 @@ devtools will only log actions from each separated store unlike in a typical _co
 You can log a specific action type for each `set` function by passing a third parameter:
 
 ```jsx
-const createBearSlice = (set, get) => ({
-  eatFish: () =>
-    set(
-      (prev) => ({ fishes: prev.fishes > 1 ? prev.fishes - 1 : 0 }),
-      false,
-      'bear/eatFish'
-    ),
-})
+const useBearStore = create(devtools((set) => ({
+  ...
+  eatFish: () => set(
+    (prev) => ({ fishes: prev.fishes > 1 ? prev.fishes - 1 : 0 }),
+    false,
+    'bear/eatFish'
+  ),
+  ...
 ```
 
 You can also log the action's type along with its payload:
 
 ```jsx
-const createBearSlice = (set, get) => ({
-  addFishes: (count) =>
-    set((prev) => ({ fishes: prev.fishes + count }), false, {
-      type: 'bear/addFishes',
-      count,
-    }),
-})
+  ...
+  addFishes: (count) => set(
+    (prev) => ({ fishes: prev.fishes + count }),
+    false,
+    { type: 'bear/addFishes', count, }
+  ),
+  ...
```

Comment:
Hi @dai-shi, I have issue when chaining persist type here : 
![image](https://github.com/pmndrs/zustand/assets/8638967/8120b753-1c8e-44ba-98ea-396c34c2fa67)
![image](https://github.com/pmndrs/zustand/assets/8638967/7da4d550-7cc1-4f3c-88e4-178ad71f3c7d)
this chaining type does not allow 3 params

Reply:
Isn't it because of `getMiddlewares` typing?
Doesn't it work if you put `persist` directly there?
You might want to open a new discussion with a typescript playground reproduction.

Reply:
can you check this [link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgEwKYDcYQgGwM4A0cYqUewe8AvnAGZQQhwBEAXgK6UCGAdsgPQhgyZDlQB3LlFTMA3AChQkWIjgBlGFxioAwtK3QiG6KgCy7TVigBJNDxjBawUkQDG+7XBr1GLDtz45eXlUAA9leFcIHko4AHNUGAARDCxcPDgAXjgACh4uEFQALjhKKGAeOIBKEpgATxIIWhRU7HwsgD5c2iIIMAdovCrOlsw2vByexHk4OAA6Bb6BmIIZ0qtiuHzC1dntzYByAEEAGROAfTUAFQB5ACUAUQP5KirgsIi4KJj4BJgABVI5Fi2TyBU2ZQq1VqDVQTWIQIo8EyXUmvX6wEGwxRCLISLR01mCzmS0xKzW+xK+xebxC4WgkUGv0SpmEogkUlQGVBlPW5Uq2K6AB41lciGtTGAMmFtHwMgBtYzScyWaC2VD2RzOKBEdg8ADWPAg4h4AF15aasnALeLZqZXNLQrLkAqlWYLAYbHYHE4XHA9YbjWaLVaLfJUbQShotLoPNAhWLraaiPa8B0aupLLHUJ6EympUR5fKDv5NHx+GgxukDkQeBhSMn5gtU6aujiEGtvrFzpW0u1sn8UlX8GDCm9Zl34OcSHiQfFEoDZzBR6hx18mXBpHh2DhkXAe610jlp4jKDk6HAiEh9l4qmvpDB2FAeJuuTv4FwMgGjSa4J-MzGeg5lYeZwJKhDWsWpa8AIvbjDWWz1lAjbEi2HQvME9QkHAOhaG6VodrMTh4AAFlyVLsCAABGpAKLMwEAGIUCRRQ5FUKLoBAwgKFQ7z0iok7+nguh4RsVruMBqBCrhMBuh0bE5GsfysiIYiSFuOQHK4WgHFUOQ5MJMCCrkhGzHQzHkXAACcuxmYxzElGxnRrGZpSJIprmuQZWbGTkSDEWReAlNw2hzAFXJwF0ACMcAAPzrDGYUWRkAC0cAxSUAAMt62Z5dBcPgqC5Z5zD2aRzDFbMVS2a8azVfIVSyEAA) ?

Reply:
https://tsplay.dev/w2Q5zW
It's tricky but it's how Mutator works.

Reply:
oh, cast type, I'm also using it to work around this